### PR TITLE
fix: blank sidepane remains open after disabling chat, vb, peerlist

### PR DESCRIPTION
**Pull request by KaustubhKumar05**
- [x] Test in customiser for side effects

### CODE REVIEW

File: packages/roomkit-react/src/Prebuilt/components/AppData/useSidepaneResetOnLayoutUpdate.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  sidepaneOption: string,
+) => {
+  const { elements } = useRoomLayoutConferencingScreen();
+  const sidepane = useHMSStore(selectAppData(APP_DATA.sidePane));
+  const resetSidePane = useSidepaneReset();
+  useEffect(() => {
+    if (sidepane === sidepaneOption && !elements?.[layoutKey]) {
+      resetSidePane();
+    }
+  }, [elements, elements?.[layoutKey], resetSidePane, sidepane, layoutKey, sidepaneOption]);
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/AppData/useSidepaneResetOnLayoutUpdate.tsx:21]: elements is required? elements?.[layoutKey] is not enough? if not just elements should be enough no?

File: packages/roomkit-react/src/Prebuilt/layouts/SidePane.tsx [hunk 2]
```diff
@@ -199,7 +191,15 @@ const SidePane = ({
       return null;
     });
 
-  if (!trackId && !SidepaneComponent) {
+  const resetSidePane = useSidepaneReset();
+
+  useEffect(() => {
+    return () => {
+      resetSidePane();
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/layouts/SidePane.tsx:198]: this already happens somewhere I believe? or it doesn/t?

File: packages/roomkit-react/src/Prebuilt/components/Chat/Chat.tsx [hunk 3]
```diff
@@ -42,9 +44,6 @@ export const Chat = () => {
     [hmsActions, vanillaStore],
   );
 
-  if (!elements?.chat) {
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/Chat/Chat.tsx:45]: why is this removed btw?

Comment (KaustubhKumar05) [reply packages/roomkit-react/src/Prebuilt/components/Chat/Chat.tsx:45]: Redundant code, handled by the hook uniformly across all sidepane components

File: packages/roomkit-react/src/Prebuilt/components/AppData/useSidepaneResetOnLayoutUpdate.tsx (hunk 1 above)
Comment (KaustubhKumar05) [reply packages/roomkit-react/src/Prebuilt/components/AppData/useSidepaneResetOnLayoutUpdate.tsx:21]: With just elements, the useEffect was not being triggered since the key was being set to null

File: packages/roomkit-react/src/Prebuilt/components/AppData/useSidepaneResetOnLayoutUpdate.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import { selectAppData, useHMSStore } from '@100mslive/react-sdk';
+import { useRoomLayoutConferencingScreen } from '../../provider/roomLayoutProvider/hooks/useRoomLayoutScreen';
+// @ts-ignore
+import { useSidepaneReset } from './useSidepane';
+import { APP_DATA } from '../../common/constants';
+
+// Closes the sidepane if an element is removed from the layout via the customiser
+export const useSidepaneResetOnLayoutUpdate = (
+  layoutKey: keyof DefaultConferencingScreen_Elements,
+  sidepaneOption: string,
```
[the remark this answers is not in the document]
Comment (KaustubhKumar05) [reply packages/roomkit-react/src/Prebuilt/components/AppData/useSidepaneResetOnLayoutUpdate.tsx:12]: Will update

**Review by raviteja83 — APPROVED** [review]
